File: JS/203-remove-linked-list-elements.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @param {number} val
 * @return {ListNode}
 */
var removeElements = function(head, val) {
    let header = new ListNode(_, head);
    let current = header;
    while (current.next) {
        if (current.next.val == val) {
            current.next = current.next.next;
        } else {
            current = current.next;
        }
    }
    head = header.next;
    header.next = null;
    return head;
};

File: C++/203-remove-linked-list-elements.cpp
#include "header.h"
 
class Solution {
public:
    ListNode* removeElements(ListNode* head, int val) {
        ListNode* header  = new ListNode(0);
        header->next = head;
        ListNode* current = header;
        while (current->next != nullptr) {
            if (current->next->val == val) {
                ListNode* temp = current->next;
                current->next = temp->next;
                delete temp;
            } else {
                current = current->next;
            }
        }
        head = header->next;
        delete header;
        return head;
    }
};